src/seamless_communication/inference/transcriber.py

```diff
@@ -316,10 +328,40 @@ def transcribe(
             decoded_audio["waveform"].size(0) / decoded_audio["sample_rate"]
         )
 
-        return self.run_inference(
-            src,
-            src_lang,
-            length_seconds,
-            filter_width,
-            sequence_generator_options,
-        )
+        if length_seconds > chunk_size_sec:
+
+            waveform_2d = decoded_audio.get("waveform")
+            waveform_1d = decoded_audio.get("waveform").view(-1)
+            segmenter = SileroVADSegmenter(
+              sample_rate=sample_rate,
+              chunk_size_sec=chunk_size_sec,
+              pause_length=pause_length_sec,
+              )
+            segmented_audios = segmenter.segment_long_input(waveform_1d)
+            transcriptions = []
+            for start, end in segmented_audios:
+                segment = waveform_2d[start:end, :]
+                src_segment = self.convert_to_fbank(
+                    {"waveform": segment, "sample_rate": decoded_audio.get("sample_rate"), 
+                     "format": decoded_audio.get("format")})["fbank"]
+                print(src_segment.device)
+                length_seconds_segment = segment.size(0) / sample_rate
+                transcription_segment = self.run_inference(
+                    src_segment,
+                    src_lang,
+                    length_seconds_segment,
+                    filter_width,
+                    sequence_generator_options,
+                )
+                transcriptions.append(str(transcription_segment))
+
+            return " ".join(transcriptions)
+        else:
+
+            return self.run_inference(
```

Comment:
nit: to avoid having two copies of this call, can you implement this function differently: 
1) materialized src_sgments (for short audios it will be an array with one element - the original audio)
2) a loop with calls for each src_segment
3) fainal join

Reply:
fixed